cloudify_cli/commands/sites.py

```diff
@@ -0,0 +1,162 @@
+@cfy.options.latitude
+@cfy.options.longitude
+@cfy.options.visibility(mutually_exclusive_required=False)
+@cfy.options.tenant_name(required=False, resource_name_for_help='site')
+@cfy.options.common_options
+@cfy.assert_manager_active()
+@cfy.pass_client(use_tenant_in_header=True)
+@cfy.pass_logger
+def create(name, latitude, longitude, visibility, tenant_name, client, logger):
+    """Create a new site
+
+    `NAME` is the new site's name
+    """
+    utils.explicit_tenant_name_message(tenant_name, logger)
+    validate_visibility(visibility)
+    client.sites.create(name, latitude, longitude, visibility)
+    logger.info('Site `{0}` created'.format(name))
+
+
+@sites.command(name='get', short_help='Get details for a single site')
+@cfy.argument('name', callback=cfy.validate_name)
+@cfy.options.tenant_name(required=False, resource_name_for_help='site')
+@cfy.options.common_options
+@cfy.assert_manager_active()
+@cfy.pass_client(use_tenant_in_header=True)
+@cfy.pass_logger
+def get(name, tenant_name, client, logger):
+    """Get details for a single site
+
+    `NAME` is the site's name
+    """
+    utils.explicit_tenant_name_message(tenant_name, logger)
+    graceful_msg = 'Requested site with name `{0}` was not found in this ' \
+                   'tenant'.format(name)
+    with handle_client_error(404, graceful_msg, logger):
+        logger.info('Getting info for site `{0}`...'.format(name))
+        site_details = client.sites.get(name)
+        print_single(SITES_COLUMNS, site_details, 'Requested site info:')
+
+
+@sites.command(name='update', short_help='Update an existing site')
+@cfy.argument('name', callback=cfy.validate_name)
+@cfy.options.latitude
+@cfy.options.longitude
+@cfy.options.update_visibility
+@cfy.options.new_name('site')
+@cfy.options.tenant_name(required=False, resource_name_for_help='site')
+@cfy.options.common_options
+@cfy.assert_manager_active()
+@cfy.pass_client(use_tenant_in_header=True)
+@cfy.pass_logger
+def update(name, latitude, longitude, visibility, new_name, tenant_name,
+           client, logger):
+    """Update an existing site
+
+    `NAME` is the site's name
+    """
+    utils.explicit_tenant_name_message(tenant_name, logger)
+    validate_visibility(visibility)
+    graceful_msg = 'Requested site with name `{0}` was not found'.format(name)
+    with handle_client_error(404, graceful_msg, logger):
+        client.sites.update(name, latitude, longitude, visibility, new_name)
+        logger.info('Site `{0}` updated'.format(name))
+
+
+@sites.command(name='list', short_help='List all sites')
+@cfy.options.sort_by('name')
+@cfy.options.descending
+@cfy.options.common_options
+@cfy.options.tenant_name_for_list(required=False,
+                                  resource_name_for_help='site')
+@cfy.options.all_tenants
+@cfy.options.search
+@cfy.options.pagination_offset
+@cfy.options.pagination_size
+@cfy.assert_manager_active()
+@cfy.pass_client()
+@cfy.pass_logger
+def list(sort_by,
```

Comment:
Maybe in the future, we will want to sort by some lat/long/geo way

Reply:
You can pass the name of the filed to sort.
But yes, if it is something more complicate of geo we will need to add it.